app/src/main/java/br/com/catbag/gifreduxsample/ui/GifListActivity.java

```diff
@@ -39,11 +38,25 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
+    protected void onStart() {
+        super.onStart();
+        mActionCreator.loadGifs();
+    }
+
+    @Override
+    public boolean hasStateChanged(AppState newState, AppState oldState) {
+        return !newState.equals(oldState);
+    }
+
+    @Override
     public void onStateChanged(AppState appState) {
-        if (!appState.getGifs().isEmpty()) {
+        if (!appState.getGifs().isEmpty() && mGifProgressVisibility) {
             mGifProgressVisibility = false;
+            anvilRender();
```

Comment:
Yes that is the idea! In futures maintenance it`s better to have the things working without too much effort!